lib/accessly/policy/base.rb

```diff
@@ -267,6 +267,7 @@ def _action_defined?(action_name)
 
       def self._actions
         @@_actions ||= {}
+        @@_actions[self.name] ||= {}
```

Comment:
This is a little mind bending... but I *think* what we can do here to solve this problem would be to use "class instance variables" instead of "class variables" and then we would not have to worry about the name space! One of the big differences are that the `@@class_variable` is accessible through the inheritance chain but will all refer to the same thing variable! (so they overwrite themselves) where as with the `@class_instance_variable` it sets that value for just that instance of `Class` (since everything is ruby is an object and the new class is just an instance of `Class`). Here is a quick example to try an illustrate:
```Ruby
class Testing
  def self.something=(val); @something= val; end
  def self.another=(val); @@another=val;end
  def self.something; @something; end
  def self.another; @@another;end
end
class First < Testing;end
class Second < Testing;end
First.something
Second.something
First.something = "something first"
First.something
Second.something
First.another = "another first"
First.another
Second.another
Second.another = "another second"
Second.another
First.another
Second.something = "something second"
Second.something
First.something
```

I should have picked better var names there lol, but you get the idea 😆 

I tried googling for a better explanation than I am giving here, and this is the best I came up with: https://rubymonk.com/learning/books/4-ruby-primer-ascent/chapters/45-more-classes/lessons/113-class-variables

What do you thing? It seems to me that switching to using "class instance variables" would allow us to prevent each class from overwriting each other without the additional overhead of namespacing them? Additionally... it would keep the global namespace clear of namespaced vars from other classes?

Reply:
@rreinhardt9 yes! my initial concern is that the we would need to create an instance of the class in order to set the variable, but that doesn't seem to be the case (Places where we call class functions on a policy without creating an object that access `_actions_on_objects`)

@armilam - is there anything we could be missing to make use of this ?

Reply:
I admit that I get these concepts confused, but only in Ruby... :shakes_fist:

I think what Ross is saying makes sense, and I think it's worth trying!

And to be clear, since we would be referencing an "instance variable" from a class context (that is, inside a method defined with `self.` prefixing the name), then the "instance" in that case would be the class itself. And therefore, we shouldn't have to create an object instance of the class.

Yeah, confusing... but I just might work. And if it does, then it means the `Accessly::Policy::Base` class wouldn't be storing actions for all policy classes. Instead, each policy class would be storing its own actions.

Reply:
So Ross's suggestion worked, eh? Nice!